app/src/main/java/org/simple/clinic/drugs/selection/custom/CustomDrugEntryUiRenderer.kt

```diff
@@ -22,6 +22,9 @@ class CustomDrugEntryUiRenderer(
         setSheetTitle(model.drugName, model.dosage, model.drugFrequencyToFrequencyChoiceItemMap[model.frequency]!!.label)
 
       showDefaultDosagePlaceholder(model.dosage, model.dosageHasFocus)
+
+      if (model.isSaveButtonInProgressState)
+        ui.showSaveButtonProgressState()
```

Comment:
Just an FYI, if there is no else condition we can just have it in a single line.